File: app/src/main/java/com/micewine/emu/core/RunServiceClass.java
package com.micewine.emu.core;

import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class RunServiceClass {
    private Context ctx;
    private Intent service;

    public void runService(Class service, Context ctx) {
        this.ctx = ctx;
        this.service = new Intent(ctx, service);
        ctx.startService(this.service);
    }

    public void stopService() {
        ctx.stopService(this.service);
    }
}

File: app/src/main/java/com/micewine/emu/core/RunServiceClass.kt
package com.micewine.emu.core

import android.content.Context
import android.content.Intent

class RunServiceClass {
    private var ctx: Context? = null
    private var service: Intent? = null
    fun runService(service: Class<*>?, ctx: Context?) {
        this.ctx = ctx
        this.service = Intent(ctx, service)
        ctx!!.startService(this.service)
    }

    fun stopService() {
        if (service != null) {
            ctx!!.stopService(service)
        }
    }
}
